api/client/search.go

```diff
@@ -60,9 +75,26 @@ func (cli *DockerCli) CmdSearch(args ...string) error {
 	w := tabwriter.NewWriter(cli.out, 10, 1, 3, ' ', 0)
 	fmt.Fprintf(w, "NAME\tDESCRIPTION\tSTARS\tOFFICIAL\tAUTOMATED\n")
 	for _, res := range results {
+		// --automated and -s, --stars are deprecated since Docker 1.13
 		if (*automated && !res.IsAutomated) || (int(*stars) > res.StarCount) {
 			continue
 		}
+		if filterOptions["is-automated"] == "true" && !res.IsAutomated {
+			continue
+		}
+		if filterOptions["is-automated"] == "false" && res.IsAutomated {
+			continue
+		}
+		if filterOptions["is-official"] == "true" && !res.IsOfficial {
+			continue
+		}
+		if filterOptions["is-official"] == "false" && res.IsOfficial {
+			continue
+		}
```

Comment:
to make this section a bit less repetitive maybe a boolean can be defined, e.g. ` is{Flagname} := filterOptions["{flagname}"] == "true"`, then instead of 4 `if`'s it becomes 2: "`if is{Flagname} != res.Is{Flagname} { continue }`"